clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -349,19 +347,20 @@ private void sendProduceRequest(long now, int destination, short acks, int timeo
             recordsByPartition.put(tp, batch);
         }
 
-        ProduceRequest produceRequest = new ProduceRequest(acks, timeout, produceRecordsByPartition);
-        RequestHeader header = this.client.nextRequestHeader(ApiKeys.PRODUCE);
+        ProduceRequest.Builder requestBuilder =
+                new ProduceRequest.Builder(acks, timeout, produceRecordsByPartition);
```

Comment:
It sounds like we would be in the clear then, since we're sending an ApiVersionsRequest followed by a follow-on request?